bootloader/src/old_machine_bootloader_windows.c

```diff
@@ -0,0 +1,275 @@
+    }
+}
+
+int ping_island(int argc, char * argv[])
+{
+    // Get all machine IP's
+    int addrCount = 0;
+    char* hostname;
+    char** IPs = getIpAddresses(&addrCount, &hostname);
+    printf("hostname: %s\n", hostname);
+    printf("Addr Count: %d", addrCount);
+    char* IPstring = concatenate(addrCount, IPs, "\", \"");
+    printf("Concatenated ips: %s\n", IPstring);
+
+    char* windowsVersion = getOsVersion();
+    printf("Windows version: %s\n", windowsVersion);
+
+    // Find which argument is tunnel flag
+    int i, tunnel_i=0, server_i=0;
+    for(i=1;i<argc;i++)
+    {
+        if(strcmp(argv[i], "-t") == 0 || strcmp(argv[i], "--tunnel") == 0){
+            tunnel_i = i+1;
+        } else if(strcmp(argv[i], "-s") == 0 || strcmp(argv[i], "--server") == 0){
+            server_i = i+1;
+        }
+    }
+
+    // Form request struct
+    struct requestData reqData;
+    reqData.osVersion = windowsVersion;
+    reqData.hostname = hostname;
+    reqData.IPstring = IPstring;
+	reqData.tunnel = NULL;
+	reqData.glibcVersion = NULL;
+
+    int request_failed = 1;
+    // Convert server argument string to wchar_t
+	if (server_i == 0) {
+		error("Server argument not passed\n");
+	}
+	char* server = (char*)malloc((strlen(argv[server_i]) + 1));
+	wchar_t* serverW;
+
+	char* responseFormat = "{\"system\":\"%s\", \"os_version\":\"%s\", \"hostname\":\"%s\", \"tunnel\":%s, \"ips\": [\"%s\"]}";
+  	char* systemStr = "windows";
+  	char* requestContents;
+  	wchar_t* requestContentsW;
+    if (server_i != 0) {
+        server = replaceSubstringOnce(argv[server_i], ISLAND_SERVER_PORT, "");
+		serverW = (wchar_t*)malloc(sizeof(wchar_t) * (strlen(server) + 1));
+		mbstowcs_s(NULL, serverW, strlen(server) + 1, server, strlen(server) + 1);
+
+        requestContents = getRequestDataJson(reqData, responseFormat, systemStr);
+        requestContentsW = (wchar_t*)malloc(sizeof(wchar_t) * (strlen(requestContents) + 1));
+        mbstowcs_s(NULL, requestContentsW, strlen(requestContents) + 1, requestContents, strlen(requestContents) + 1);
+        printf("Sending request\n");
+        request_failed = sendRequest(serverW, NULL, requestContentsW);
+    }
+
+    // Convert tunnel argument string to wchar_t
+    if (tunnel_i != 0 && serverW != NULL && request_failed) {
+        wchar_t* tunnel = (wchar_t*)malloc(sizeof(wchar_t) * (strlen(argv[tunnel_i])+1));
+        mbstowcs_s(NULL, tunnel, strlen(argv[tunnel_i])+1, argv[tunnel_i], strlen(argv[tunnel_i]));
+        wprintf(L"Tunnel: %s\n", tunnel);
+        reqData.tunnel = argv[tunnel_i];
+
+        requestContents = getRequestDataJson(reqData, responseFormat, systemStr);
+        requestContentsW = (wchar_t*)malloc(sizeof(wchar_t) * (strlen(requestContents) + 1));
+        mbstowcs_s(NULL, requestContentsW, strlen(requestContents) + 1, requestContents, strlen(requestContents) + 1);
+        request_failed = sendRequest(serverW, tunnel, requestContentsW);
+    }
+    return request_failed;
```

Comment:
You return request_failed rather than if the bootloader should continue.

Reply:
So far I treated  it as the same thing. Should monkey run if bootloader failed to communicate?

Reply:
Moved windows logic to the bootloader, even though I don't like this decision.

Reply:
I don't understand what windows logic means @VakarisZ